one_shot_scripts/update_deck_format.js

```diff
@@ -0,0 +1,74 @@
+require('dotenv').config();
+
+// eslint-disable-next-line import/extensions
+const mongoose = require('mongoose');
+const Deck = require('../models/deck.js');
+const carddb = require('../serverjs/cards');
+const cubefn = require('../serverjs/cubefn');
+
+const batchSize = 100;
+
+let basics;
+let basicsList;
+
+const migrateDeck = async (deck) => {
+  if (deck.cards && deck.cards.length !== 0) {
+    return null;
+  }
+  const cards = [];
+  for (const seat of deck.seats || []) {
+    seat.deck = (seat.deck || []).map((pack) =>
+      pack.map((card) => {
+        cards.push(card);
+        return cards.length - 1;
+      }),
+    );
+    seat.sideboard = (seat.sideboard || []).map((pack) =>
+      pack.map((card) => {
+        cards.push(card);
+        return cards.length - 1;
+      }),
+    );
+    seat.pickorder = (seat.pickorder || []).map((card) => cards.findIndex((card2) => card.cardID === card2.cardID));
+  }
+  deck.cards = cards.concat(basicsList);
+  return deck;
```

Comment:
needs better error handling. There is so much errored data across the DB, check out the old deck migration script for reference. I need to check a lot of fields to see if they were objects, primitives, or arrays all the time

Reply:
I'm going to strip out the migrations and just do it on the fly.